src/static-map.js

```diff
@@ -451,241 +377,81 @@ export default class MapGL extends Component {
+      this.props.onClick(latLong, pos);
     }
 
     if (this.props.onClickFeatures) {
-      // Radius enables point features, like marker symbols, to be clicked.
-      const size = this.props.clickRadius;
-      const bbox = [[pos[0] - size, pos[1] - size], [pos[0] + size, pos[1] + size]];
-      const features = this._map.queryRenderedFeatures(bbox, this._queryParams);
+      const features = this._getFeatures({pos, radius: this.props.clickRadius});
       if (!features.length && this.props.ignoreEmptyFeatures) {
         return;
       }
       this.props.onClickFeatures(features);
     }
   }
 
-  @autobind _onZoom({pos, scale}) {
-    const point = new Point(...pos);
-    const transform = cloneTransform(this._map.transform);
-    const around = unprojectFromTransform(transform, point);
-    transform.zoom = transform.scaleZoom(this._map.transform.scale * scale);
-    transform.setLocationAtPoint(around, point);
-    this._callOnChangeViewport(transform, {isDragging: true});
-  }
-
-  @autobind _onZoomEnd() {
-    this._callOnChangeViewport(this._map.transform, {isDragging: false});
-  }
-
   render() {
     const {className, width, height, style} = this.props;
-    const mapStyle = {
-      ...style,
-      width,
-      height,
-      cursor: this._getCursor()
-    };
 
-    let content = [
-      <div key="map" ref="mapboxMap"
-        style={ mapStyle } className={ className }/>,
-      <div key="overlays" className="overlays"
-        style={ {position: 'absolute', left: 0, top: 0} }>
-        { this.props.children }
-      </div>
-    ];
-
-    if (this.state.isSupported && this.props.onChangeViewport) {
-      content = (
-        <MapInteractions
-          onMouseDown ={ this._onMouseDown }
-          onMouseDrag ={ this._onMouseDrag }
-          onMouseRotate ={ this._onMouseRotate }
-          onMouseUp ={ this._onMouseUp }
-          onMouseMove ={ this._onMouseMove }
-          onMouseClick = { this._onMouseClick }
-          onTouchStart ={ this._onTouchStart }
-          onTouchDrag ={ this._onTouchDrag }
-          onTouchRotate ={ this._onTouchRotate }
-          onTouchEnd ={ this._onTouchEnd }
-          onTouchTap = { this._onTouchTap }
-          onZoom ={ this._onZoom }
-          onZoomEnd ={ this._onZoomEnd }
-          width ={ this.props.width }
-          height ={ this.props.height }>
-
-          { content }
-
-        </MapInteractions>
-      );
-    }
+    const mapContainerStyle = {...style, width, height, position: 'relative'};
+    const mapStyle = {...style, width, height};
+    const overlayContainerStyle = {position: 'absolute', left: 0, top: 0};
 
+    // Note: a static map still handles clicks and hover events
     return (
-      <div
-        style={ {
-          ...this.props.style,
-          width: this.props.width,
-          height: this.props.height,
-          position: 'relative'
-        } }>
-
-        { content }
+      <div style={mapContainerStyle}>
+
+        <EventManager
```

Comment:
I think there is something odd around the `StaticMap` wrapping an `EventManager` here. I understand the want is to expose hover/click events, but perhaps we shouldn't even do that. A `StaticMap` should just take in a set of mapbox-gl compatible props and render the map.

One other reason to perhaps take out the EventManager here is to lessen the layers when someone is trying to render an `InteractiveMap`. Right now that one render of `InteractiveMap` would look like this (to my understanding):
```
MapControls -> EventManager -> EventManager -> MapGL
```